cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -1425,6 +1425,51 @@ TEST(GroupBy, MinMaxDecimal) {
   }
 }
 
+TEST(GroupBy, MinOrMax) {
+  auto table =
+      TableFromJSON(schema({field("argument", float64()), field("key", int64())}), {R"([
+    [1.0,   1],
+    [null,  1]
+                        ])",
+                                                                                    R"([
+    [0.0,   2],
+    [null,  3],
+    [4.0,   null],
+    [3.25,  1],
```

Comment:
Extend test case to get min/max between `NaN, Inf, null, value`.

Reply:
Done.